ddl/sequence_test.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ddl_test
+
+import (
+	. "github.com/pingcap/check"
+	"github.com/pingcap/parser/auth"
+	"github.com/pingcap/parser/model"
+	"github.com/pingcap/parser/mysql"
+	"github.com/pingcap/tidb/session"
+	"github.com/pingcap/tidb/util/testkit"
+)
+
+var _ = Suite(&testSequenceSuite{&testDBSuite{}})
+
+type testSequenceSuite struct{ *testDBSuite }
+
+func (s *testSequenceSuite) TestCreateSequence(c *C) {
+	s.tk = testkit.NewTestKit(c, s.store)
+	s.tk.MustExec("use test")
+	s.tk.MustGetErrCode("create sequence `seq  `", mysql.ErrWrongTableName)
+
+	// maxvalue should be larger than minvalue.
+	_, err := s.tk.Exec("create sequence seq maxvalue 1 minvalue 2")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[ddl:4136]Sequence 'test.seq' values are conflicting")
+
+	// maxvalue should be larger than minvalue.
+	_, err = s.tk.Exec("create sequence seq maxvalue 1 minvalue 1")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[ddl:4136]Sequence 'test.seq' values are conflicting")
+
+	// maxvalue shouldn't be equal to MaxInt64.
+	_, err = s.tk.Exec("create sequence seq maxvalue 9223372036854775807 minvalue 1")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[ddl:4136]Sequence 'test.seq' values are conflicting")
+
+	// TODO : minvalue shouldn't be equal to MinInt64.
```

Comment:
Why not implement it?

Reply:
Int64 from `-9223372036854775808 ~ 9223372036854775807`
by now, parser parse a negative number use the rule ` '-' NUM ` and action like `$$ = -$2.(int64)`, so here is a mistake, `$2` won't get to the abs(-9223372036854775808).

by the way, do you have some good ideas about this?

Reply:
Get it. Later I will fix the parser.